conformance/src/main/scala/za/co/absa/enceladus/conformance/interpreter/rules/NegationRuleInterpreter.scala

```diff
@@ -78,7 +81,8 @@ case class NegationRuleInterpreter(rule: NegationConformanceRule) extends RuleIn
       case _: ShortType => when(inputColumn === Short.MinValue, errorColumnUDF).otherwise(null)
       case _: IntegerType => when(inputColumn === Int.MinValue, errorColumnUDF).otherwise(null)
       case _: LongType => when(inputColumn === Long.MinValue, errorColumnUDF).otherwise(null)
-      case _ => null
+      case a => throw new IllegalArgumentException("NegationRuleInterpreter.getError() should be called only for " +
```

Comment:
Why this change? Due to the nested checking this branch is unreachable anyway. And if reached, will halt the whole conformance at runtime in the middle of processing (which is somewhat late anyway). 🤔

Reply:
Bad error message from this part of code is, what I think stopped me and I needed to ask Ruslan for help. I vote we keep the message. Otherwise, you get unreadable messages.

Reply:
Exactly. That branch should be unreachable. But when a new data type is added and the function returns null the error message we get doesn't make any sense. It is a NullPointerException somewhere deep in DeepArrayTransformation. Double deep 😄

Reply:
Shouldn't there be an error message there then instead of crashing the whole process?

Reply:
Agreed killing the processing is good in this case of _"should never happen"_ 😉